gateway/analytics.go

```diff
@@ -309,7 +244,7 @@ func (r *RedisAnalyticsHandler) recordWorker() {
 				record.RawPath = "/" + record.RawPath
 			}
 
-			if encoded, err := msgpack.Marshal(record); err != nil {
+			if encoded, err := r.analyticsSerializer.Encode((*analytics.AnalyticsRecord)(record)); err != nil {
```

Comment:
is this casting necessary?

Reply:
yes, there's a missmatch of type. Being record of type *AnalyticsRecord as "github.com/TykTechnologies/tyk-pump/analytics".AnalyticsRecord. We can discuss it, I know what is the underlying issue